src/main/java/com/ibm/cloudant/kafka/connect/CloudantSinkTask.java

```diff
@@ -48,8 +45,10 @@ public class CloudantSinkTask extends SinkTask {
 	private int taskNumber;
 	public static String guid_schema = null;
 	private Boolean replication;
-	public static volatile JSONArray jsonArray = new JSONArray();
 
+	private List<Map<String, Object>> jsonArray = new ArrayList<>();
```

Comment:
As you say, I suspect we will have much bigger performance issues if we start processing such large numbers of records.

However, `LinkedList` is probably appropriate since we would never expect to perform random access on this list.

Reply:
Used LinkedList in 7585eb7